tests/behat/includes/FeatureContext.php

```diff
@@ -0,0 +1,224 @@
+<?php
+/**
+ * Class FeatureContext.
+ *
+ * Feature tests context class with plugin-specific steps.
+ */
+
+namespace WordPress\Plugin_Check\Behat_Utils;
+
+use Behat\Behat\Hook\Scope\AfterFeatureScope;
+use Behat\Behat\Hook\Scope\AfterScenarioScope;
+use Behat\Behat\Hook\Scope\BeforeFeatureScope;
+use Behat\Behat\Hook\Scope\BeforeScenarioScope;
+use WP_CLI\Process;
+use WP_CLI\Tests\Context\FeatureContext as WP_CLI_FeatureContext;
+use WP_CLI\Utils;
+use function WP_CLI\Tests\Context\wp_cli_behat_env_debug;
+
+/**
+ * Feature tests context class with plugin-specific steps.
+ *
+ * This class extends the one that is provided by the wp-cli/wp-cli-tests package.
+ * To see a list of all recognized step definitions, run `vendor/bin/behat -dl`.
+ */
+final class FeatureContext extends WP_CLI_FeatureContext {
+
+	/**
+	 * The current feature.
+	 *
+	 * @var \Behat\Gherkin\Node\FeatureNode|null
+	 */
+	private static $feature;
+
+	/**
+	 * The current scenario.
+	 *
+	 * @var \Behat\Gherkin\Node\ScenarioInterface|null
+	 */
+	private $scenario;
+
+	/**
+	 * @BeforeFeature
+	 */
+	public static function store_feature( BeforeFeatureScope $scope ) {
+		self::$feature = $scope->getFeature();
+	}
+
+	/**
+	 * @BeforeScenario
+	 */
+	public function store_scenario( BeforeScenarioScope $scope ) {
+		$this->scenario = $scope->getScenario();
+	}
+
+	/**
+	 * @AfterScenario
+	 */
+	public function forget_scenario( AfterScenarioScope $scope ) {
+		$this->scenario = null;
+	}
+
+	/**
+	 * @AfterFeature
+	 */
+	public static function forget_feature( AfterFeatureScope $scope ) {
+		self::$feature = null;
+	}
+
+	/**
+	 * @Given a WP install(ation) with the Plugin Check (plugin)
+	 */
+	public function given_a_wp_installation_with_plugin_check() {
+		$this->install_wp();
+
+		// Symlink the current project folder into the WP folder as a plugin.
+		$project_dir = realpath( self::get_vendor_dir() . '/../' );
+		$plugin_dir  = $this->variables['RUN_DIR'] . '/wp-content/plugins';
+		$this->ensure_dir_exists( $plugin_dir );
+		$this->proc( "ln -s {$project_dir} {$plugin_dir}/plugin-check" )->run_check();
+
+		// Activate the plugin.
+		$this->proc( 'wp plugin activate plugin-check' )->run_check();
+	}
+
+	/**
+	 * @When /^I (run|try) the WP-CLI command `([^`]+)`$/
+	 */
+	public function when_i_run_the_wp_cli_command( $mode, $command ) {
+		$command = "wp {$command}";
+
+		$with_code_coverage = getenv( 'BEHAT_CODE_COVERAGE' );
+		if ( \in_array( $with_code_coverage, [ true, 'true', 1, '1' ], true ) ) {
+			$command = "{$command} --require={PROJECT_DIR}/tests/behat/utils/maybe-generate-wp-cli-coverage.php";
```

Comment:
Does this work when there is an existing `--require` argument? That would be relevant for runtime checks due to the current limitations of having to require the early logic.

Reply:
Yes, WP-CLI supports multiple `--require` arguments.

Also worth noting that this kind of `--require` usage should become obsolete anyway with #153

Reply:
@felixarntz Please let me know if this answers your question :-)

Reply:
Thanks, LGTM!